pkgs/os-specific/linux/kernel/generic.nix

```diff
@@ -126,7 +130,13 @@ let
     depsBuildBuild = [ buildPackages.stdenv.cc ];
     nativeBuildInputs = [ perl gmp libmpc mpfr ]
       ++ lib.optionals (lib.versionAtLeast version "4.16") [ bison flex ]
-      ++ lib.optional (lib.versionAtLeast version "5.2") pahole;
+      ++ lib.optional (lib.versionAtLeast version "5.2") pahole
+      ++ lib.optionals (lib.versionAtLeast version "6.3") [ rust-bindgen rustfmt rustc ]
+    ;
+
+    env = lib.optionalAttrs (lib.versionAtLeast version "6.3") {
+      RUST_LIB_SRC = rustPlatform.rustLibSrc;
+    };
```

Comment:
I don't feel too strongly, but isn't it confusing to have some environment variables passed through `env`, and others passed through in the root derivation attrset?